### ISSUE TITLE
Cop idea: Style/BitPredicates

### ISSUE THREAD

**Issue by zverok** [rubocop/rubocop#13050]
## Is your feature request related to a problem? Please describe.

Ruby 2.5 (!) [introduced](https://rubyreferences.github.io/rubychanges/2.5.html#allbits-anybits-nobits) `Integer#anybits?`/`#allbits?`/`#nobits?` predicates. Accordingly, it would be cool if Rubocop suggested those corrections:
```ruby
# 1. anybits?
# bad
(variable & MY_FLAGS).positive?
# good
variable.anybits?(MY_FLAGS)

# 2. allbits?
# bad
(variable & MY_FLAGS) == MY_FLAGS
# good
variable.allbits?(MY_FLAGS)

# 3. nobits?
# bad
(variable & MY_FLAGS).zero?
# good
variable.nobits?(MY_FLAGS)
```

I’ve seen a lot of codebases that use the bit-math version, and in many cases, predicates version seems to be clearer. It also has an educational value: I constantly see people writing bit-math to check flags seven years after methods introduction (because they simply don’t remember methods exist).

Haven’t yet time enough to work on the cop myself (well, I happen to have a draft branch, and I feel guilty for never finishing it, maybe I’ll do... but it is what it is) :cry:

**Comment by vlad-pisanov** [comment]
One can also access individual bits via `[]`, perhaps it could be part of such a cop as well.

Bit extraction:
```ruby
# bad
(x >> bit) & 1

# good
x[bit]
```

Bit checking:
```ruby
# bad
x & (1 << bit) > 0

# good
x[bit] == 1

# bad
x & (1 << bit) == 0

# good
x[bit] == 0
```

Bit checking against constant powers of 2:
```ruby
# bad
(x & 8).positive?
(x & 0b01000).positive?
(x & (1 << 3)).positive?
(x & 8) == 8

# good
x[3] == 1
```
